libqtile/backend/wayland/window.py

```diff
@@ -161,6 +164,8 @@ def mapped(self, mapped: bool) -> None:
         else:
             self.core.mapped_windows.remove(self)
         self.core.stack_windows()
+        if hasattr(self, "_idle_inhibitors") and len(self._idle_inhibitors) > 0:
+            self.core.check_idle_inhibitor()
```

Comment:
Is this because of another class inheriting this method, which might not have this attribute? If so, this hack will fortunately not be necessary soon by having more well-defined types: https://github.com/qtile/qtile/pull/3293

Reply:
Hmm that would solve this, it is present on several places in this PR, should I wait until your change is merged, or should I leave it as is and change it later?

Reply:
I'm happy to merge this first and rebase that other PR on top of this. However, we do need to account for the different window classes, specifically we need to ensure that if `Window.cmd_static` is called to turn an XDG shell window into a `Static` window, that window's `_idle_inhibitors` also needs to be copied to the new object. Similarly we need to register idle inhibitors that are created for windows represented by `Static` and handle those just the same if the window is part of the layer shell or xdg shell. If it makes things easier to see how these things are represnted then waiting until that other PR is merged might make sense. Up to you.

Reply:
Ok, I've added idle_inhibitors_count to other window classes, and to cmd_static. It seems to work fine.